### PR TITLE
Add documentation about scoped cloudflare api tokens

### LINKED ISSUE #49: DNS01: Document Cloudflare API token usage
Our Cloudflare DNS01 guide is lacking any detail, and only contains a YAML snippet currently: https://github.com/cert-manager/website/blob/master/content/en/docs/configuration/acme/dns01/cloudflare.md

We should update this document to be a bit clearer, as well as documenting the newly introduce token based authentication (including what permissions the token requires in order to be usable).

There's more information on what permissions are required here: https://github.com/jetstack/cert-manager/pull/2170#issuecomment-561237219

/good-first-issue
/help
/area acme/dns01

### PR DESCRIPTION
**Pull request by terrabitz**
This PR adds some additional info about creating and using Cloudflare API tokens in ACME DNS01 challenges. 

Closes #49

### CODE REVIEW

**Comment by munnerz** [comment]
Thanks for these additions 😄 looks good to me!

/lgtm
/ok-to-test
/approve

File: content/en/docs/configuration/acme/dns01/cloudflare.md [hunk 1]
```diff
@@ -5,6 +5,69 @@ weight: 30
 type: "docs"
 ---
 
+To use Cloudflare, you may use one of two types of tokens. **API Tokens** allow application-scoped keys bound to specific zones and permissions, while **API Keys** are globally-scoped keys that carry the same permissions as your account.
```
Comment (munnerz) [inline content/en/docs/configuration/acme/dns01/cloudflare.md]: I think other areas of the docs use "CloudFlare", hence why "Cloudflare" is being flagged for incorrect spelling!

Comment (terrabitz) [reply content/en/docs/configuration/acme/dns01/cloudflare.md]: Alright, I (hopefully) just fixed the two spelling errors it caught.

**Pull request merged**